src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/ReportService.kt

```diff
@@ -182,4 +184,34 @@ class ReportService(
       }
     }.getOrNull()
   }
+
+  @Transactional
+  fun changeReportStatus(id: UUID, changeStatusRequest: ChangeStatusRequest): Pair<ReportWithDetails, Boolean>? {
+    return reportRepository.findById(id).map {
+      val changed = it.status != changeStatusRequest.newStatus
+      if (changed) {
+        // TODO: determine which transitions are allowed
+
+        val now = LocalDateTime.now(clock)
+        val user = authenticationFacade.getUserOrSystemInContext()
+        it.changeStatus(
+          newStatus = changeStatusRequest.newStatus,
+          changedAt = now,
+          changedBy = user,
+        )
+        it.modifiedAt = now
+        it.modifiedBy = user
+      }
+
+      it.toDtoWithDetails().apply {
+        if (changed) {
```

Comment:
Couldn't you move this inside the `if (changed)` block above and avoid this other `if`?

It's not much about the if itself but I think it makes sense to keep the two inside the same block as they're only happening together anyway (same condition)